riscv/zicfiss.h

```diff
@@ -0,0 +1,29 @@
+#ifndef _RISCV_ZICFISS_H
+#define _RISCV_ZICFISS_H
+
+#define SET_XSSE(xSSE) \
+  reg_t prv = STATE.prv; \
+  if (prv == PRV_M) \
+    xSSE = false; \
+  else if (prv == PRV_S || prv == PRV_HS) \
+    xSSE = (get_field(STATE.menvcfg->read(), MENVCFG_SSE) == 1); \
+  else if (p->extension_enabled('S')) \
+    xSSE = (get_field(STATE.senvcfg->read(), SENVCFG_SSE) == 1); \
+  if (prv == PRV_S && STATE.v) \
+    xSSE = xSSE & (get_field(STATE.henvcfg->read(), HENVCFG_SSE) == 1);
+
+#define check_ssp(reg, ssp) \
+  reg_t shadow_return_addr; \
+  if (xlen == 32) \
+    shadow_return_addr = MMU.ss_load<uint32_t>(ssp); \
+  else \
+    shadow_return_addr = MMU.ss_load<uint64_t>(ssp); \
+  software_check(reg == shadow_return_addr, SHADOW_STACK_FAULT);
+
```

Comment:
I think the name `check_ssp` may be misleading because it does not check the `ssp`. How about `check_return_address`?